### PR TITLE
Refactor many request assert for limit tests

### PR DESCRIPTION
**Pull request by jsmolar**

### CODE REVIEW

File: testsuite/httpx/__init__.py [hunk 1]
```diff
@@ -65,6 +65,15 @@ def __str__(self):
         return f"Result[error={self.error}]"
 
 
+class ResultList(list):
+    """List-like object for Result"""
+
+    def assert_all(self, status_code=None):
```
Comment (averevki) [inline testsuite/httpx/__init__.py:71]:
```suggestion
    def assert_all(self, status_code):
```

**Review by averevki — COMMENTED** [review]
What about `ResultOne` and `ResultMany`?

**Comment by pehala** [comment]
> What about `ResultOne` and `ResultMany`?

I think `ResultList` is a better name

File: testsuite/httpx/__init__.py [hunk 2]
```diff
@@ -65,6 +65,15 @@ def __str__(self):
...
+        """Assert all responses that contain certain status code"""
+        for request in self:
+            assert request.status_code == status_code, f"Rate Limited resource unexpectedly rejected request: {request}"
```
Comment (pehala) [inline testsuite/httpx/__init__.py:74]: I would improve the error message wording and to also include index of the request that is wrong

File: testsuite/httpx/__init__.py (hunk 1 above)
Comment (jsmolar) [reply testsuite/httpx/__init__.py:71]: agree, we can add named parameters if/when more asserts is needed.

File: testsuite/httpx/__init__.py [hunk 3]
```diff
@@ -65,6 +65,18 @@ def __str__(self):
...
 
+class ResultList(list):
+    """List-like object for Result"""
+
+    def assert_all(self, status_code):
+        """Assert all responses that contain certain status code"""
+        for request in self:
+            assert request.status_code == status_code, (
+                f"Status code assertion failed for request {self.index(request)+1} out of {len(self)} requests: "
+                f"{request} != {status_code}"
```
Comment (azgabur) [inline testsuite/httpx/__init__.py:76]:
```suggestion
                f"{request.status_code} != {status_code}"
```

File: testsuite/tests/kuadrant/limitador/test_route_rule.py [hunk 4]
```diff
@@ -33,10 +33,16 @@ def rate_limit(rate_limit):
 
 def test_rule(client):
     """Tests that RLP correctly applies to the given HTTPRoute rule"""
-    responses = client.get_many("/get", 5)
-    assert all(
-        r.status_code == 200 for r in responses
-    ), f"Rate Limited resource unexpectedly rejected requests {responses}"
+    responses = client.get_many("/get", 7)
```
Comment (azgabur) [inline testsuite/tests/kuadrant/limitador/test_route_rule.py:36]: Why the change to 7? The limit is set to 5.

File: testsuite/httpx/__init__.py (hunk 3 above)
Comment (azgabur) [reply testsuite/httpx/__init__.py:76]: If the response status code can be seen in the error message than its okay

File: testsuite/tests/kuadrant/limitador/test_route_rule.py (hunk 4 above)
Comment (azgabur) [reply testsuite/tests/kuadrant/limitador/test_route_rule.py]: Also in the function `test_rule_missmatch`

Comment (jsmolar) [reply testsuite/tests/kuadrant/limitador/test_route_rule.py]: `test_rule_missmatc` is new test method where 7 is intended

**Review by azgabur — APPROVED** [review]

**Review by pehala — APPROVED** [review]
